tools/logger/logger.c

```diff
@@ -150,7 +150,7 @@ static int append_filter_config(struct convert_config *config, const char *input
 
```

Comment:
@marc-hb can you explain why do we need this commit?

Reply:
this commit turn on `printf` like formatting string vs argument list check in compilation time for `asprintf` and `log_err` functions

Reply:
The `printf` attribute finds run-time memory corruption at compile time, always seemed very useful to me. It works with clang too BTW (tested).